### PR TITLE
rename camlpstreams to camlp-streams

### PR DESCRIPTION
**Pull request by avsm**
- remove dune constraint from dune-package (automatically added)
- rename findlib module to camlp_streams and opam to camlp-streams
- remove (version) entry, as dune-release will added that upon
  cutting a release archive.

Committing the generated opam file will also let `opam pin add`
work for the development repository (with `--dev`)

### CODE REVIEW

File: src/dune [hunk 1]
```diff
@@ -1,6 +1,6 @@
-(library (name camlpstreams) (public_name camlpstreams)
+(library
+ (name camlp_streams)
+ (public_name camlp-streams)
  (modules Stream Genlex)
```
Comment (trefis) [inline src/dune]: AFAICT this line is not necessary.

Comment (xavierleroy) [reply src/dune]: Which line?

**Review by xavierleroy — APPROVED** [review]
Thanks for all the updates.  I'm a little lost between dashes and underscores, but if all the tools involved are happy, I'm happy too.

File: src/dune (hunk 1 above)
Comment (avsm) [reply src/dune]: The `modules` line I think. bdb3abc4e7acce08df5a26ae70252133d7fc5751

**Comment by avsm** [comment]
It's essentially a historical artefact that we're slowly homogenising. There are three classes of names:

- **module names**: which have to obey OCaml conventions. `dune` uses the `(name)` field as the toplevel wrapped module name by default, so it restricts libraries to valid OCaml module names by default.
- **findlib archive names**: in the case of this library, the modules are _unwrapped_ and so in theory dune could allow dashes in archive names, since the modules themselves are valid OCaml syntax. Need to double check if ocamlobjinfo and friends work ok with cmx/a files with hyphens. I've filed https://github.com/ocaml/dune/issues/4783 to track this.
- **opam package**: the `public_name` gives the whole package a name, which can be used to separate files at the opam level (and this can contain multiple findlib archives). The easy (but I agree unintuitive) fix is to simply name the archive `(name)` with underscores, and the `public name` with dashes. https://github.com/ocaml/dune/issues/4783 also suggests a better error hint in this case.

Unfortunately, all three namespaces are used in various existing systems (consider findlib subpackages for example), so the migration itself is an ongoing process. The defaults in dune shove users towards matching findlib and opam names, so that `opam install <pkg>` is also what the user would type into `(library <pkg>)` to depend on it.  (thanks @jeremiedimino for clarifying some of this for me)

**Pull request merged**